PortalSwift/Sources/PortalSwift/Portal.swift

```diff
@@ -0,0 +1,939 @@
+//
+//  Portal.swift
+//
+//  Created by Portal Labs, Inc.
+//  Copyright © 2022 Portal Labs, Inc. All rights reserved.
+//
+
+import AuthenticationServices
+import Foundation
+import Mpc
+
+/// The main Portal class.
+public class Portal {
+  @available(*, deprecated, renamed: "addresses", message: "Please use the async getter for `addresses`")
+  public var address: String? {
+    self.keychain.legacyAddress
+  }
+
+  public var addresses: [PortalNamespace: String?] {
+    get async throws {
+      try await self.keychain.getAddresses()
+    }
+  }
+
+  public var client: ClientResponse? {
+    get async throws { try await self.api.client }
+  }
+
+  public var chainId: Int? {
+    self.provider.chainId
```

Comment:
Should this be `chainId` still or should we call it `chainReference`?

Reply:
This is only to retain legacy support for `chainId`, so it will be removed in the future. I'll mark this as deprecated also, actually.